JobRunner/config.py

```diff
@@ -34,7 +35,7 @@ def _get_admin_token():
 class Config:
     def __init__(self, workdir=None, base_url=None, job_id=None, use_ee2=True):
         self.job_id = job_id
-        self.base_url = "https://ci.kbase.us/services/"
+        self.base_url = os.environ.get(_KB_BASE_URL, "https://ci.kbase.us/services/")
```

Comment:
No action required: I put this into chatgpt and this is what it gave on how to refactor this code
```
import logging
import os
from socket import gethostname

logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

class Config:
    DEFAULT_BASE_URL = "https://ci.kbase.us/services/"
    DEFAULT_MAX_TASKS = 10
    ENV_TOKEN = "KB_AUTH_TOKEN"
    ENV_ADMIN_TOKEN = "KB_ADMIN_AUTH_TOKEN"
    ENV_BASE_URL = "KB_BASE_URL"
    ENV_DEBUG = "JOBRUNNER_DEBUG_MODE"
    ENV_MAX_TASKS = "JR_MAX_TASKS"
    ENV_RUNTIME = "RUNTIME"

    def __init__(self, workdir=None, base_url=None, job_id=None, use_ee2=True):
        self.job_id = job_id
        self.base_url = os.getenv(self.ENV_BASE_URL, self.DEFAULT_BASE_URL)
        self.workdir = workdir or os.getcwd()
        self.debug = os.getenv(self.ENV_DEBUG, "").lower() == "true"
        self.runtime = os.getenv(self.ENV_RUNTIME, "docker")
        self.max_tasks = int(os.getenv(self.ENV_MAX_TASKS, self.DEFAULT_MAX_TASKS))
        self.token = self._get_token()
        self.admin_token = self._get_admin_token()
        self.hostname = gethostname()
        self._setup_urls(use_ee2)
        os.makedirs(self.workdir, exist_ok=True)

    def _get_token(self, env_var=ENV_TOKEN, fallback_file="token"):
        token = os.getenv(env_var)
        if not token and os.path.exists(fallback_file):
            with open(fallback_file) as file:
                token = file.read().strip()
                os.environ[env_var] = token
        return token

    def _get_admin_token(self):
        token = os.getenv(self.ENV_ADMIN_TOKEN)
        if token:
            os.environ.pop(self.ENV_ADMIN_TOKEN, None)  # Attempt to remove after retrieval
            if self.ENV_ADMIN_TOKEN in os.environ:
                logger.error("Failed to sanitize environment: admin token still present")
        return token

    def _setup_urls(self, use_ee2):
        auth_ext = "auth/api/legacy/KBase/Sessions/Login"
        self.ee2_url = f"{self.base_url}ee2" if use_ee2 else None
        self.catalog_url = f"{self.base_url}catalog"
        self.auth_url = f"{self.base_url}{auth_ext}"
        self.auth2_url = f"{self.base_url}auth/api/V2/token"
```

Reply:
The suggested approach appears much cleaner, but for now, I guess we can just stick with the current working code.